### ISSUE TITLE
Add user preference dashboard for notifications

### ISSUE THREAD

**Issue by agjohnson** [readthedocs/readthedocs.org#10882]
We started to add some user preferences for notifications recently, but we also talked about moving these to a dedicated UI page and expanding some of the options users have on that page.

Currently, there are a few deprecation specific notification options on the main profile page form. These can actually be removed as the deprecation has passed. These options are specific to single notifications.

What we may be aiming for is instead of giving the user opt out for a single notification, we want to make the opt in/out more generalized, so users can opt out of more notifications and/or future notifications.

We haven't dug too deep into what we want exactly here, but a few ideas were:

- Use Sendgrid to manage opt-out? I believe there are some native features that help us here. We might need to use internal modeling either way though.
- Standard email opt out options might be the best pattern, which would be to give the user message classes to opt out from -- "marketing", "suggestions", "notices", etc.

- [ ] Drop existing notification preferences, they aren't use anymore afaik
- [ ] Add some layer for opt out modeling -- continue using internal modeling or use Sendgrid for opt out
- [ ] Add UI for opt in/out -- form, view, etc in user preferences dashboard

**Comment by humitos** [comment]
This is out of the scope of the new notification design document I'm working on at #10890, but if we go in that direction and implement it, we can easily add a new field to the `Notification` model to track this and decide whether or not show/list/send notification based on user's preferences.

```python
# Group defined for opt-out via notification preferences
group = models.CharField(choides=[MARKETING, SUGGESTIONS, NOTICES])
```

This issue is more related to email notification in my opinion, where the new notification system is more based on site notifications. We can eventually adapt the new notification system and use it for emails as well, but I don't want to start _that_ big and focus on our main goal for now.

**Comment by agjohnson** [comment]
> This issue is more related to email notification in my opinion

Yup, this is exclusively for email notifications I feel. I don't think we need on site notifications to be configurable.
